grin/src/server.rs

```diff
@@ -209,15 +215,11 @@ impl Server {
 	pub fn start_miner(&self, config: pow::types::MinerConfig) {
 		let cuckoo_size = global::sizeshift();
 		let proof_size = global::proofsize();
-		let net_adapter = self.net_adapter.clone();
 
 		let mut miner = miner::Miner::new(config.clone(), self.chain.clone(), self.tx_pool.clone());
 		miner.set_debug_output_id(format!("Port {}", self.config.p2p_config.unwrap().port));
 		thread::spawn(move || {
-			let secs_5 = time::Duration::from_secs(5);
-			while net_adapter.syncing() {
-				thread::sleep(secs_5);
-			}
+			// TODO - sync always runs, so need to consider how the miner interacts with the sync process
```

Comment:
Miner may build blocks on the current chain view, which should be immediately orphaned and can cause confusion. So better not to run the miner when we're not close to the latest.

Reply:
So I'm pretty sure we can do the same thing what we're doing with `update_head` where we compare the head of the header chain to the head of the block chain.
And if there's too much difference then we're in "sync/catchup" mode and we should pause the miner loop (may still sit mining the current block until the 90s timeout).